qa/workunits/rbd/cli_migration.sh

```diff
@@ -124,8 +128,56 @@ EOF
 
     rbd migration abort ${dest_image}
 
+    # no snap name or snap id
+    expect_false rbd migration prepare --import-only \
+        --source-spec "{\"type\": \"native\", \"pool_id\": ${pool_id}, \"image_name\": \"${base_image}\"}" \
+        ${dest_image}
+
+    # invalid source spec JSON
+    expect_false rbd migration prepare --import-only \
+        --source-spec "{\"type\": \"native\", \"pool_id\": ${pool_id}, \"image_name\": \"${base_image}\", \"snap_name\": non-existing}" \
+        ${dest_image}
+
+    # non-existing snap name
+    expect_false rbd migration prepare --import-only \
+        --source-spec "{\"type\": \"native\", \"pool_id\": ${pool_id}, \"image_name\": \"${base_image}\", \"snap_name\": \"non-existing\"}" \
+        ${dest_image}
+
+    # invalid snap name
+    expect_false rbd migration prepare --import-only \
+        --source-spec "{\"type\": \"native\", \"pool_id\": ${pool_id}, \"image_name\": \"${base_image}\", \"snap_name\": 123456}" \
+        ${dest_image}
+
+    # non-existing snap id passed as int
+    expect_false rbd migration prepare --import-only \
+        --source-spec "{\"type\": \"native\", \"pool_id\": ${pool_id}, \"image_name\": \"${base_image}\", \"snap_id\": 123456}" \
+        ${dest_image}
+
+    # non-existing snap id passed as string
+    expect_false rbd migration prepare --import-only \
+        --source-spec "{\"type\": \"native\", \"pool_id\": ${pool_id}, \"image_name\": \"${base_image}\", \"snap_id\": \"123456\"}" \
+        ${dest_image}
+
+    # invalid snap id
+    expect_false rbd migration prepare --import-only \
+        --source-spec "{\"type\": \"native\", \"pool_id\": ${pool_id}, \"image_name\": \"${base_image}\", \"snap_id\": \"foobar\"}" \
+        ${dest_image}
+
+    # snap id passed as int
+    local snap_id=$(rbd snap ls ${base_image} --format xml | xmlstarlet sel -t -v "//snapshots/snapshot[name='2']/id")
```

Comment:
Don't we prefer JSON output and jq parsing?

Reply:
Definitely, but this function already has a similar `xmlstarlet` incantation for fetching the pool ID.  If these tests were added in a new function or if I needed something more than to extract a single field, I would have went with `jq`.  As it is, I didn't want to have both tools used in same function for exactly the same thing.